pulser-core/pulser/sequence/_schedule.py

```diff
@@ -252,6 +253,50 @@ def __iter__(self) -> Iterator[_TimeSlot]:
 class _DMMSchedule(_ChannelSchedule):
     detuning_map: DetuningMap
 
+    def get_samples(
+        self,
+        ignore_detuned_delay_phase: bool = True,
+        qubits: dict[QubitId, np.ndarray] | None = None,
+    ) -> DMMSamples:
+        ch_samples = super().get_samples(
+            ignore_detuned_delay_phase=ignore_detuned_delay_phase
+        )
+        init_fields = {
+            f.name: getattr(ch_samples, f.name)
+            for f in fields(ch_samples)
+            if f.init
+        }
+        if qubits is None:
+            raise ValueError(
+                "'qubits' must be defined when extracting the samples of a"
+                " DMM channel."
+            )
+        return DMMSamples(
+            **init_fields, detuning_map=self.detuning_map, qubits=qubits
+        )
+
+    def insert_slm(self, duration: int, det: float) -> None:
+        """Inserts a constant pulse as first operation."""
+        new_slot = [self.slots[0]]
+        new_slot.append(
+            _TimeSlot(
+                Pulse.ConstantAmplitude(0, ConstantWaveform(duration, det), 0),
+                0,
+                duration,
+                self.slots[0].targets,
+            )
+        )
+        for slot in self:
+            new_slot.append(
+                _TimeSlot(
+                    slot.type,
+                    slot.ti + duration,
+                    slot.tf + duration,
+                    slot.targets,
+                )
+            )
```

Comment:
So, if I already have things in this `DMMSchedule`, which are likely aligned with what's in the other channels, won't this operation screw up that alignment?